poetry/core/factory.py

```diff
@@ -94,7 +97,12 @@ def configure_package(
         package.classifiers = config.get("classifiers", [])
 
         if "readme" in config:
-            package.readme = root / config["readme"]
+            if isinstance(config["readme"], (list, tuple)):
```

Comment:
```suggestion
            if isinstance(config["readme"], Sequence):
```

I would much prefer to see `Sequence` from `collections.abc` here.

Reply:
I can't do that! Because `str` is also a `Sequence`. I elaborate below.

To not break compatibility, I need to handle `type(readme) == 'str'` to wrap the single readme file in a single `list`/`tuple`.

With the changes, both `readme` declarations are possible now:
```toml
[tool.poetry]
# str syntax (keep compatibility)
readme = "README.md"
# list syntax (extension)
readme = [
    "README.md"
]
```

Reply:
A little suggestion: I noticed TOML parser always returns a `list`-like object, so I could make the code more readable by simply do
```python
if isinstance(config["readme"], list):
```

Reply:
I think I'd go the other direction, and check for `str` since Python 3 does not have multiple string types.

Reply:
What do you mean? To invert the `if` to check if is a `str` first?

Reply:
I see what you meant now, it was late night, I needed a sleep :smile: